modules/catalog-dsl/src/test/java/org/apache/ignite/internal/catalog/sql/CreateTableTest.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.catalog.sql;
+
+import static org.apache.ignite.catalog.ColumnType.INTEGER;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.is;
+
+import org.apache.ignite.catalog.IndexType;
+import org.apache.ignite.catalog.Options;
+import org.junit.jupiter.api.Test;
+
+class CreateTableTest {
+    @Test
+    void ifNotExists() {
+        String sql = createTable().ifNotExists().name("table1").addColumn("col1", INTEGER).toSqlString();
+        assertThat(sql, is("CREATE TABLE IF NOT EXISTS table1 (col1 int);"));
+
+        // quote identifiers
+        sql = createTableQuoted().ifNotExists().name("table1").addColumn("col1", INTEGER).toSqlString();
+        assertThat(sql, is("CREATE TABLE IF NOT EXISTS \"table1\" (\"col1\" int);"));
+    }
+
+    @Test
+    void names() {
+        String sql = createTable().name("public", "table1").addColumn("col1", INTEGER).toSqlString();
+        assertThat(sql, is("CREATE TABLE public.table1 (col1 int);"));
+
+        sql = createTable().name("", "table;1--test\n\r\t;").addColumn("col1", INTEGER).toSqlString();
```

Comment:
Thaaaat is reeeeally interesting. How do we define that "table;1--test\n\r\t;" will became "table1"???

Reply:
Good point, maybe we should fail the query in these case?